### ISSUE TITLE
Pytest with coverage 5.0 fails: AttributeError: 'NoneType' object has no attribute '_getframe'

### ISSUE THREAD

**Issue by jakeleventhal** [coveragepy/coveragepy#893]
Pinning coverage to <5.0 doesn't throw error.

```
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-4.6.7, py-1.8.0, pluggy-0.13.1
rootdir: /usr/src
plugins: forked-1.1.3, ordering-0.6, cov-2.8.1, xdist-1.30.0
collected 11 items
tests/unit_tests/test_service.py ......                                   [ 54%]
tests/unit_tests/test_backup_service.py .....                             [100%]
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/_pytest/main.py", line 206, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/_pytest/main.py", line 250, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/pluggy/manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/pytest_cov/plugin.py", line 254, in pytest_runtestloop
INTERNALERROR>     self.cov_controller.finish()
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/pytest_cov/engine.py", line 136, in summary
INTERNALERROR>     total = self.cov.report(**options)
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/coverage/control.py", line 840, in report
INTERNALERROR>     report_contexts=contexts, skip_empty=skip_empty,
INTERNALERROR>   File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
INTERNALERROR>     return self.gen.next()
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/coverage/control.py", line 53, in override_config
INTERNALERROR>     cov.config = cov.config.copy()
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/coverage/config.py", line 333, in copy
INTERNALERROR>     return copy.deepcopy(self)
INTERNALERROR>   File "/usr/lib/python2.7/copy.py", line 190, in deepcopy
INTERNALERROR>     y = _reconstruct(x, rv, 1, memo)
INTERNALERROR>   File "/usr/lib/python2.7/copy.py", line 334, in _reconstruct
INTERNALERROR>     state = deepcopy(state, memo)
INTERNALERROR>   File "/usr/lib/python2.7/copy.py", line 163, in deepcopy
INTERNALERROR>     y = copier(x, memo)
INTERNALERROR>   File "/usr/lib/python2.7/copy.py", line 257, in _deepcopy_dict
INTERNALERROR>     y[deepcopy(key, memo)] = deepcopy(value, memo)
INTERNALERROR>   File "/usr/lib/python2.7/copy.py", line 190, in deepcopy
INTERNALERROR>     y = _reconstruct(x, rv, 1, memo)
INTERNALERROR>   File "/usr/lib/python2.7/copy.py", line 329, in _reconstruct
INTERNALERROR>     y = callable(*args)
INTERNALERROR>   File "/usr/lib/python2.7/collections.py", line 69, in __init__
INTERNALERROR>     self.__update(*args, **kwds)
INTERNALERROR>   File "/usr/lib/python2.7/_abcoll.py", line 564, in update
INTERNALERROR>     if isinstance(other, Mapping):
INTERNALERROR>   File "/usr/lib/python2.7/abc.py", line 144, in __instancecheck__
INTERNALERROR>     return cls.__subclasscheck__(subtype)
INTERNALERROR>   File "/usr/lib/python2.7/abc.py", line 180, in __subclasscheck__
INTERNALERROR>     if issubclass(subclass, scls):
INTERNALERROR>   File "/usr/lib/python2.7/abc.py", line 180, in __subclasscheck__
INTERNALERROR>     if issubclass(subclass, scls):
INTERNALERROR>   File "/usr/local/lib/python2.7/dist-packages/typing.py", line 1400, in __subclasscheck__
INTERNALERROR>     if (sys._getframe(1).f_globals['__name__'] in whitelist or
INTERNALERROR> AttributeError: 'NoneType' object has no attribute '_getframe'
```

**Comment by nedbat** [comment]
Thanks for the exotic error report! Can you give me specific instructions to reproduce the issue?

**Comment by jakeleventhal** [comment]
I wish I had more info to help recreate the problem. Unfortunately, I'm unable to reproduce it locally. The issue is only in CI system (Bamboo).

**Comment by nedbat** [comment]
@jakeleventhal Is this error still happening for you? Have you been able to try the latest versions of coverage.py?

**Comment by jakeleventhal** [comment]
Yes, it is still happening on coverage==5.0.3

**Comment by nedbat** [comment]
@jakeleventhal Thanks. Even though it doesn't reproduce locally, if you could link us to the repo and testing instructions, we might be able to figure something out.

**Comment by jakeleventhal** [comment]
This seems resolved on 5.3? Not sure why but marking as closed for now
